citest/base/base_test_case.py

```diff
@@ -123,41 +126,63 @@ def __end_step_context(self, relation):
       return
     JournalLogger.end_context(relation=relation)
 
-  def __call__(self, result=None):
+  def __trap_skip(self, delegate, test, reason):
+    self.__final_outcome_relation = 'VALID'
+    return delegate(test, reason)
+
+  def __trap_success(self, delegate, test):
+    self.__final_outcome_relation = 'VALID'
+    return delegate(test)
+
+  def __trap_failure(self, delegate, test, err):
+    self.__final_outcome_relation = 'INVALID'
+    return delegate(test, err)
+
+  def __trap_error(self, delegate, test, err):
+    self.__final_outcome_relation = 'ERROR'
+    error_details = '%s: %s' % (err[0], err[1])
+    JournalLogger.journal_or_log_detail(
```

Comment:
What's the reasoning for logging on error only? Would we want to log on failure as well?

Reply:
I think failures would already be logged because of the reporting built into the framework.
The errors are more likely to occur outside the framework thus not be logged.